core/shared/src/main/scala/zio/logging/LogFilter.scala

```diff
@@ -232,10 +232,60 @@ object LogFilter {
     val mappingsSorted = mappings.map(splitNameByDotAndLevel.tupled).sorted(nameLevelOrdering)
     val nameGroup      = group.map(splitNameByDot)
 
+    @tailrec
+    def compareRoutes(l: List[String], m: List[String]): Boolean = (l, m) match {
```

Comment:
not sure if there will be noticeable performance difference
but maybe it could be good to try implementation with traversing both list with one iterator each

basically in similar way like `List.startsWith` is implemented

because in this code it is look like there may be a lot of unnecessary iterations over both list

Reply:
I think I have another idea. When creating the LogFilter it should already separate glob-like filters from simple ones. Then have separate implementations to avoid runtime penalties for the simple filters and the glob-like filter can then be implemented more efficient.

Reply:
Hmm, I also see the matcher argument is public so probably make it optional. Then when there is no need for glob-like filters there is also no performance loss.

Reply:
I will make changes to this PR somewhere in de next days.